### PR TITLE
Upgrade to React 16

### PR DESCRIPTION
**Pull request by puffo**
Hi there!

This upgrades the project to React 16 (#189).

Enzyme 3 requires that we re-find elements that might have changed (see [this discussion](https://github.com/airbnb/enzyme/issues/1221)). It has resulted in some of the tests looking a bit clunkier in that we unfortunately can't instantiate all the necessary variables at the start of the tests.

I'll need some help to finish it up though - in attempting to merge the latest changes from #195, I can't seem to get the most recent test passing. 

Any suggestions/help?

### CODE REVIEW

**Comment by patw0929** [comment]
Thank you for this PR! I've leave some comment above.
Hmmm... however I have no idea about the failed tests too.
Need some time to figure out.

**Comment by ignatiusreza** [comment]
Hi guys, sorry for bumping this issue.. anything I can do to help move this PR forward? would be nice to have this library be react 16 compatible, since it is currently the last package holding us from upgrading to react 16..

**Comment by puffo** [comment]
Hey @ignatiusreza, I was stuck after updating this branch (from master) to resolve the merge conflicts. I couldn't get the changes from #195 working in the test suite.

Would you be able to try on your side?

**Comment by ignatiusreza** [comment]
managed to make it pass with react-16 https://github.com/ignatiusreza/react-intl-tel-input/tree/upgrade/react

@puffo do you mind if I open another PR?

**Comment by puffo** [comment]
Thanks for your help @ignatiusreza! I've pushed up the changes and cleaned up as per the changes on your fork.

I'm now struggling with the Travis CI build, but it looks like it the build isn't pulling from my branch. 

Not sure if you can help out here @patw0929 or @ignatiusreza ?

**Comment by patw0929** [comment]
@puffo @ignatiusreza Thank you bros! Awesome!
About the Travis CI build fail, could you refer this 92df903 to see if works?

**Comment by puffo** [comment]
Build is now green @patw0929 ! Thanks!

File: package.json [hunk 1]
```diff
@@ -22,11 +22,16 @@
...
+    "react": "^16.2.0 <17.0.0",
+    "react-dom": "^16.2.0 <17.0.0"
   },
   "dependencies": {
     "classnames": "^2.2.5",
+    "enzyme-adapter-react-16": "^1.1.1",
+    "npm": "^5.6.0",
+    "raf": "^3.4.0",
+    "react": "^16.2.0",
+    "react-dom": "^16.2.0",
```
Comment (ignatiusreza) [inline package.json]: sorry for the late comment.. but I think these dependencies should not be forced unto the users:
- `react` and `react-dom` is already mentioned as peerDependencies,
- `npm` is not used, not sure why it's there in the first place
- `enzyme-adapter-react-16` is only used for development, and user does not necessarily test their code using `enzyme`
- `raf` is not the only option for polyfilling `requestAnimationFrame`.. user might already do it through other method.. and if `react` them self doesn't force one particular way to polyfill it, I don't think that we should too..

opened a PR to fix these points: https://github.com/patw0929/react-intl-tel-input/pull/199

**Pull request merged**